src/components/Files/AudioCaptureDialog.tsx

```diff
@@ -101,10 +103,21 @@ export default function AudioCaptureDialog(props: AudioCaptureDialogProps) {
     };
   }, [show]);
 
+  const releaseMic = () => {
+    if (mediaStreamRef.current) {
+      mediaStreamRef.current.getTracks().forEach((track) => track.stop());
+      mediaStreamRef.current = null;
+    }
+  };
+
   useEffect(() => {
     if (autoRecord && show && status === "RECORDING") {
       handleStartRecording();
     }
+
+    return () => {
+      mediaStreamRef.current && releaseMic();
+    };
```

Comment:
line 107 already checks if `mediaStreamRef.current` is truthy or not. is redundant checks necessary?

Also why not just simply inline the releaseMic function here?
```suggestion
    return () => {
      if (mediaStreamRef.current) {
        mediaStreamRef.current.getTracks().forEach((track) => track.stop());
        mediaStreamRef.current = null;
      }
    };
```